### ISSUE TITLE
Inconsistant Key marshalling for increment and read/write

### ISSUE THREAD

**Issue by droberts84** [redis-store/redis-activesupport#23]
I'm using a Hash object as my key, and cannot read a key after incrementing.  It looks like the Hash key is being serialized differently in the increment function.  This is with Rails 4.1.6 and Redis-Store 4.0

See below:

    Rails.cache.increment({hkey: 'test'})
    # => 1
    redis-cli:  KEYS *
    # "{:hkey=>\"test\"}"
    Rails.cache.read({hkey: 'test'})
    # => nil
    Rails.cache.write({hkey: test}, 10)
    # => OK
    redis-cli: KEYS *
    #  "hkey=test"
    #  "{:hkey=>\"test\"}"
    Rails.cache.increment({hkey: 'test'})
    # => 2
    Rails.cache.read({hkey: 'test'})
    # => 10

**Comment by raldred** [comment]
Any sign of this ever getting merged?

**Comment by radar** [comment]
@raldred According to my helpful IRC bot:

```
[07:51:29]  <Radar>	!8ball
[07:51:30]  <helpa>	It is certain.
```

However I don't see a PR for it. Could you please submit one or link me to one if it exists already?

Thank you.

**Comment by droberts84** [comment]
@radar https://github.com/redis-store/redis-activesupport/pull/24

**Comment by zydeco** [comment]
Same problem occurs with expire, fixed in my PR #58
